recipes/license_server.rb

```diff
@@ -70,6 +70,21 @@
   notifies :touch, 'file[splunk-marker]', :immediately
 end
 
+ruby_block 'check license overuse' do
+  block do
+    # Find the total available license quota
+    total_available_license_quota = 0
+    existing_license = Dir.glob("#{node['splunk']['home']}/etc/licenses/enterprise/*.lic*")
+    existing_license.each do |file|
+      license_file = Nokogiri::XML(File.open(file))
+      max_quota = license_file.at_xpath('/license/payload/expiration_time/text()').to_s.to_i > Time.now.to_i ? license_file.at_xpath('/license/payload/quota/text()').to_s.to_i : 0
+      total_available_license_quota += max_quota
+    end
+    total_allotted_pool_size = node.run_state['total_allotted_pool_size']
+    fail "Sum of pool sizes is #{CernerSplunk.human_readable_size total_allotted_pool_size}. Exceeds total available pool size of #{CernerSplunk.human_readable_size total_available_license_quota}." if total_allotted_pool_size > total_available_license_quota
```

Comment:
We should be able to calculate this and fail at compile time... as the point of the license master is to manage licenses, and we have the XML in memory above during compile time... I worry that failing at this point could lead to a bad time (as it's possible we haven't done the first start of Splunk yet, but failing at converge time could cause Splunk to be started without password in place through the marker restart already notified).